### PR TITLE
Issue 87:  Better support for hosts that are 64 bit and TAs that are …

### PR DESCRIPTION
**Pull request by ericeberry**
…32 bit

### CODE REVIEW

File: reference/src/client/include/sa_ta_types.h [hunk 1]
```diff
@@ -189,38 +189,41 @@ typedef struct {
 
 typedef struct {
     uint8_t iv[GCM_IV_LENGTH];
-    size_t iv_length;
+    uint64_t iv_length;
     uint8_t tag[MAX_GCM_TAG_LENGTH];
-    uint8_t tag_length;
+    uint64_t tag_length;
```
Comment (mhabrat) [inline reference/src/client/include/sa_ta_types.h:194]: Do we expect to see tag lengths that are 64-bits in size?

File: reference/src/client/include/sa_ta_types.h [hunk 2]
```diff
@@ -189,38 +189,41 @@ typedef struct {
...
+    uint64_t nonce_length;
     uint8_t tag[CHACHA20_TAG_LENGTH];
-    uint8_t tag_length;
+    uint64_t tag_length;
 } sa_unwrap_parameters_chacha20_poly1305_s;
 
 typedef struct {
-    sa_digest_algorithm digest_algorithm;
-    sa_digest_algorithm mgf1_digest_algorithm;
+    uint32_t digest_algorithm;
```
Comment (mhabrat) [inline reference/src/client/include/sa_ta_types.h:212]: What's the benefit to changing these to 32-bit unsigned integers?

**Review by mhabrat — CHANGES_REQUESTED** [review]

File: reference/src/client/include/sa_ta_types.h (hunk 1 above)
Comment (ericeberry) [reply reference/src/client/include/sa_ta_types.h:194]: No, but it was formerly a size_t which is 64 bits on a 64 bit machine and 32 bits on a 32 bit machine. This just standardizes the size for marshalling.

File: reference/src/client/include/sa_ta_types.h (hunk 2 above)
Comment (ericeberry) [reply reference/src/client/include/sa_ta_types.h:212]: Enums are typically represented as uint32_t. This is just explicitly declaring it.

File: reference/src/client/include/sa_ta_types.h (hunk 1 above)
Comment (mhabrat) [reply reference/src/client/include/sa_ta_types.h:194]: It looks like it was previously a uint8_t not a size_t. Did you mean to expand it to 64 bits?

Comment (ericeberry) [reply reference/src/client/include/sa_ta_types.h:194]: Ah it was is a typo here. In sa_types.h, the field is a size_t.

**Review by mhabrat — APPROVED** [review]

**Pull request merged**